api/client/cli.go

```diff
@@ -160,6 +162,13 @@ func NewDockerCli(in io.ReadCloser, out, err io.Writer, clientFlags *cli.ClientF
 		}
 		cli.configFile = configFile
 
+		// Set the CLI's API version
+		if ver := os.Getenv("DOCKER_API_VERSION"); ver != "" {
+			api.Version = version.Version(ver)
+		} else if configFile.APIVersion != "" {
+			api.Version = version.Version(configFile.APIVersion)
```

Comment:
I understand that it's a "power user feature", but no validation at all? not sure about that; perhaps lifting the "minimum version". If we go for *no* (or only minimum) validation, the docs should have a clear warning, i.e. "use at your own risk".

Should `docker info` show that a version override is active? (this may be helpful for the GitHub issue tracker if people added a version-override and forget about it)